arangod/Aql/Query.h

```diff
@@ -208,7 +208,7 @@ class Query {
   void releaseEngine();
 
   /// @brief return the transaction, if prepared
-  inline transaction::Methods* trx() { return _trx; }
+  virtual inline transaction::Methods* trx() { return _trx; }
```

Comment:
:+1: for `virtual inline`.

Reply:
`virtual inline` will only make sense if you have an actual object at hand, not a pointer or reference.

Reply:
ok we only have pointers afaik. Will fix